### ISSUE TITLE
Defer broadcasting state change until the last action dispatch

### ISSUE THREAD

**Issue by hungtran0203** [reactjs/redux#1613]
Dispatching an action immediately broadcasting new state to all listener should be deferred to next clock to improve performance. In complexity application, there could be a lot of listeners ( >3000) subscribed and an user behavior needs two or more consequently actions to update store state for example:
function createUser(dispatch) {
     dispatch(addUserProfileAction);
     dispatch(addUserHistoryLogAction);
     dispatch(addUser...);
}

In this case, the best thing we can do is just wait until the last action is dispatched then broadcast the change to all listeners.
It can be easy to implement by wrapping code at https://github.com/reactjs/redux/blob/master/src/createStore.js#L174-L177
into setTimeout function

**Comment by Aweary** [comment]
You can batch subscription notifications using the [redux-batched-subscribe](https://github.com/tappleby/redux-batched-subscribe) store enhancer! I think this is a great example of a problem that store enhancers solve well 👍

**Comment by markerikson** [comment]
There's several other utilities that can do batching of notifications and batching of dispatches as well.  See the [Store#change-subscription](https://github.com/markerikson/redux-ecosystem-links/blob/master/store.md#store-change-subscriptions) and [Middleware#action-grouping](https://github.com/markerikson/redux-ecosystem-links/blob/master/middleware.md#action-grouping-and-interception) sections of my Redux Ecosystem catalog.

Basically, there's plenty of ways to do this in user space, and not a good reason to do it in core.
